### PR TITLE
Discovery errors handling in watch mode

### PR DESCRIPTION
**Pull request by antooni**
This PR will:
- add a test, to ensure that no discovered.json includes errors
- clear discovered.json from errors
- change DiscoveryWatcher update function, so that it stops processing if errors where discovered
Resolves L2B-729

### CODE REVIEW

File: packages/backend/discovery/pNetwork/discovered.json [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+    {
+      "name": "Erc20Vault",
+      "address": "0x112334f50Cb6efcff4e35Ae51A022dBE41a48135",
+      "code": "https://etherscan.deth.net/address/0x112334f50Cb6efcff4e35Ae51A022dBE41a48135",
+      "upgradeability": {
+        "type": "immutable"
+      },
+      "values": {
+        "PNETWORK": "0xDffE7AC6B538B4A7Fd81c98C5fba0415d63fB132"
+      }
```
Comment (michalsidzej) [inline packages/backend/discovery/pNetwork/discovered.json:15]: Why did the order of those contracts change? 🤔

File: packages/backend/src/core/discovery/utils/config.test.ts [hunk 2]
```diff
@@ -63,4 +63,18 @@ describe('discovery config.jsonc', () => {
       expect(discovery.configHash, { extraMessage }).toEqual(configHash)
     }
   })
+
+  it('discovery.json does not include errors', async () => {
+    const configs = await configReader.readAllConfigs()
+    // debugger
```
Comment (michalsidzej) [inline packages/backend/src/core/discovery/utils/config.test.ts:69]: Hello

File: packages/backend/src/core/DiscoveryWatcher.ts [hunk 3]
```diff
@@ -69,6 +69,13 @@ export class DiscoveryWatcher {
           projectConfig,
           blockNumber,
         )
+
+        if (discovery.contracts.some((c) => c.errors !== undefined)) {
+          throw new Error(
+            `Errors occurred during discovery of ${projectConfig.name}`,
+          )
+        }
```
Comment (michalsidzej) [inline packages/backend/src/core/DiscoveryWatcher.ts:77]: Maybe it's better to retry this? Or set the retry policy of the taskQueue to never retry.

File: packages/backend/src/core/discovery/utils/config.test.ts (hunk 2 above)
Comment (antooni) [reply packages/backend/src/core/discovery/utils/config.test.ts:69]: ;p

File: packages/backend/discovery/pNetwork/discovered.json (hunk 1 above)
Comment (antooni) [reply packages/backend/discovery/pNetwork/discovered.json:15]: I think this is the artifact from the past, the order looks correct, as expected from config

**Review by michalsidzej — APPROVED** [review]

**Pull request merged**